_hackathon/checkfirst.md

```diff
@@ -1,36 +1,50 @@
+
+Moreover, custom eligibility checker web applications can be expensive and time-consuming to build. So we thought - why not create a low-cost tool that government agencies can use to create eligibility checkers/calculators quickly?
 
 #### What tech stack did you use?
 
-AWS API Gateway + Lambda hosting express.js, Math.js as the calc/logic engine
+We used React for our frontend web application, ExpressJS for our backend server, and PostgreSQL for our database. We also depend on Math.js to power our calculation/logic engine. The service is hosted on API Gateway and AWS Lambda.
 
 #### What were the key challenges you faced in building CheckFirst 
 
 **Engineering challenges:**\\
-Some deliberation had to be made over how the formulas that represent the checker/calculator had to be made. Math.js proved to be a godsend, and, as it turned out, was used by similar commercial offerings out there. Given that the project probably would not get much of a budget either, we had to plan so that we can have it survive on as much of AWS Free Tier as possible
+We’re really excited to share about this challenge because it’s such a cool problem to solve (and it comes with an elegant solution)!
+One of the most challenging parts of building CheckFirst was figuring out how to perform static analysis for the calculator evaluation engine to see if a user-defined calculator was valid. Let us explain what this means (note: this part gets a little technical!).
+
+Assume that someone has built a simple calculator which has a result that depends on a question input field. Let’s say that the calculator has only one question - “What is your age?“. And the calculator just does one operation, which is to multiply your age by 2.
+
+This is a valid calculator because the operation step (“Multiply your age by 2”) can run successfully because all of its dependencies exist. Specifically, the only dependency that the operation depends on is the question “What is your age?“.
+
+However, let’s say that someone accidentally deletes the question. Or even worse, sets the result of the operation step to depend on itself. How are we able to statically determine if the calculator is a valid one?
+
+We decided to tackle this problem by using a little graph theory. We convert our entire calculator into a graph, where each variable (operation or question) is a node on the graph, and each dependency is represented by an edge. Once we model the problem as a graphical one, we can very easily find out if the calculator is valid by checking that all dependencies exist and that there are no cycles in the graph!
 
 **Design challenges:**\\
-It is hard to show the logic in a way that a civil servant would be able to self-service and build. IF ELSE statement is not something that people get intuitively.
+Calculators by themselves are not difficult to understand. However, learning how to use a tool that builds a calculator is challenging because the process is quite meta.
```

Comment:
```suggestion
Calculators by themselves are not difficult to understand. However, learning how to use a tool that builds a calculator is challenging because the inputs are not just numbers, but also strings and arrays too. As such, the process of building the logic for the calculator can get complicated.
```